Runtime/Scripts/RGBotServerListener.cs

```diff
@@ -743,6 +775,31 @@ public void HandleClientActionRequest(long clientId, RGActionRequest actionReque
                 HandleAction(clientId, actionRequest);
             });
         }
+
+        public void SaveDataForTick(long clientId, RGTickInfoData tickInfoData, List<RGActionRequest> actions,
+            List<RGValidationResult> validations)
+        {
+            _dataCollection.SaveReplayDataInfo(clientId, new RGStateActionReplayData(
+                actions: actions.ToArray(),
+                validationResults: validations.ToArray(),
+                tickInfo: tickInfoData,
```

Comment:
@RG-nAmKcAz you probs have some insights into this - I see that our serialization for `RGTickInfoData` is a little weird. When I save this data after converting to JSON, I get the tick and scene name, but no state. Any ideas how I make sure that the entire object with the game state gets serialized?

```json
{"actions":[{"action":"PerformSkill"}],"validationResults":[{"name":"Test passed","result":0,"icon":"heart","id":"54ae616b-376c-46af-995b-1c9dae19fdfb","tick":950,"message":"Test passed","passed":true,"version":1},{"name":"Test failed","result":1,"icon":"heart","id":"d752c8e2-1c3c-4a2e-bef7-a295386f4269","tick":950,"message":"Test failed","passed":false,"version":1},{"name":"Test warning","result":2,"icon":"heart","id":"d149ce27-070d-4f38-a6a6-5b23c6d8ca96","tick":950,"message":"Test warning","passed":false,"version":1}],"error":"","tickInfo":{"tick":950,"sceneName":"BossRoom"}}
```

Reply:
Figured I'd ask if you had a quick answer - otherwise, I'll implement a serialization function that uses the RGTickInfoData ToSerialize method as part of it potentially, or at least following that pattern.

Reply:
we call `.ToString()` on this explicitly when sending remote socket data is how we're working today.

haven't dug into your code to see how you're trying to serialize the object, make sure first that you're using `JsonConvert`, if its still not working with that, then yes I'd write your own serialize method

Reply:
Your suggestions saved me! I was using the wrong JSON conversion utility, and then I ran into a looping reference issue so created a serialization method to address that. Thanks!